pulp_rpm/app/models/package.py

```diff
@@ -310,7 +310,14 @@ def createrepo_to_dict(cls, package):
             changelog_limit = settings.KEEP_CHANGELOG_LIMIT or 1
             # changelogs are listed in chronological order, grab the last N changelogs from the list
             changelogs = changelogs[-changelog_limit:]
-
+        files = getattr(package, CR_PACKAGE_ATTRS.FILES, [])
+        seen = set()
+        for path in files:
+            if path in seen:
+                log.warn(f"Duplicated file: {path})")
+            else:
+                seen.add(path)
+        files = list(seen)
```

Comment:
This would reorder them (because sets are unordered data types) which I would prefer we avoid - if we could just filter out the duplicates while retaining the current order that would be ideal.

Reply:
There should also be a few lines [here](https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/tasks/synchronizing.py#L1579-L1597) that would update the files list for saved packages.